asio/cmake/dev-mode.cmake

```diff
@@ -0,0 +1,27 @@
+include(cmake/folders.cmake)
+
+#XXX include(CTest)
+option(BUILD_TESTING "Build the testing tree." ON)
```

Comment:
I'm curious what the reason for this is.

Reply:
The `CTest` module creates lots of targets you only need if you are working with [CDash](https://cmake.org/cmake/help/book/mastering-cmake/chapter/CDash.html)

Reply:
Sure. A short comment above the option explaining that would be neat then.

Reply:
Yes, good idea!

Thank you for the review